pkg/gator/sync/test/test_test.go

```diff
@@ -0,0 +1,248 @@
+			wantReqs:     map[string]parser.SyncRequirements{},
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			// convert the test resources to unstructureds
+			var objs []*unstructured.Unstructured
+			for _, input := range tc.inputs {
+				u, err := reader.ReadUnstructured([]byte(input))
+				require.NoError(t, err)
+				objs = append(objs, u)
+			}
+
+			gotReqs, gotErrs, err := Test(objs, tc.omitManifest)
+
+			require.NoError(t, err)
+
+			if gotErrs != nil {
+				t.Errorf("got unexpected errors: %v", gotErrs)
+			}
+
+			if diff := cmp.Diff(tc.wantReqs, gotReqs); diff != "" {
+				t.Errorf("diff in missingRequirements objects (-want +got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestTest_Errors(t *testing.T) {
+	DeploymentGVK := schema.GroupVersionKind{
+		Group:   "apps",
+		Version: "v1",
+		Kind:    "Deployment",
+	}
+	ServiceGVK := schema.GroupVersionKind{
+		Group:   "",
+		Version: "v1",
+		Kind:    "Service",
+	}
+	tcs := []struct {
+		name         string
+		inputs       []string
+		omitManifest bool
+		wantErrs     map[string]error
+		err          error
+	}{
+		{
+			name: "one template having error stops requirement evaluation",
+			inputs: []string{
+				fixtures.TemplateReferential,
+				fixtures.TemplateReferentialBadAnnotation,
+			},
+			omitManifest: true,
+			wantErrs: map[string]error{
+				"k8suniqueingresshostbadannotation": fmt.Errorf("json: cannot unmarshal object into Go value of type parser.CompactSyncRequirements"),
+			},
+		},
+		{
+			name: "error if manifest not provided and omitGVKManifest not set",
+			inputs: []string{
+				fixtures.TemplateReferential,
+				toYAMLString(fakes.ConfigFor([]schema.GroupVersionKind{ServiceGVK, DeploymentGVK})),
+			},
+			wantErrs: map[string]error{},
+			err:      fmt.Errorf("no GVK manifest found; please provide a manifest enumerating the GVKs supported by the cluster"),
+		},
+	}
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			// convert the test resources to unstructureds
+			var objs []*unstructured.Unstructured
+			for _, input := range tc.inputs {
+				u, err := reader.ReadUnstructured([]byte(input))
+				require.NoError(t, err)
+				objs = append(objs, u)
+			}
+
+			gotReqs, gotErrs, err := Test(objs, tc.omitManifest)
+
+			if tc.err != nil {
+				if tc.err.Error() != err.Error() {
+					t.Errorf("error mismatch: want %v, got %v", tc.err, err)
+				}
+			} else if err != nil {
+				require.NoError(t, err)
+			}
+
+			if gotReqs != nil {
+				t.Errorf("got unexpected requirements: %v", gotReqs)
+			}
+
+			for key, wantErr := range tc.wantErrs {
+				if gotErr, ok := gotErrs[key]; ok {
+					if wantErr.Error() != gotErr.Error() {
+						t.Errorf("error mismatch for %s: want %v, got %v", key, wantErr, gotErr)
+					}
+				} else {
+					t.Errorf("missing error for %s", key)
+				}
+			}
+		})
+	}
+}
+
+func toYAMLString(obj runtime.Object) string {
+	yaml, _ := yaml.Marshal(obj)
+	return string(yaml)
+}
```

Comment:
```suggestion
func toYAMLString(t *testing.T, obj runtime.Object) string {
        t.Helper()
        
	yaml, err := yaml.Marshal(obj)
	require.NoError(t, err)
	
	return string(yaml)
}
```

here's what I had in mind; then you'd have pass `t` at call sites